modules/promotion/src/Plugin/Commerce/PromotionOffer/PromotionOfferInterface.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Drupal\commerce_promotion\Plugin\Commerce\PromotionOffer;
+
+use Drupal\commerce_order\AdjustableInterface;
+use Drupal\commerce_price\Price;
+use Drupal\Component\Plugin\ConfigurablePluginInterface;
+use Drupal\Component\Plugin\PluginInspectionInterface;
+use Drupal\Core\Executable\ExecutableInterface;
+use Drupal\Core\Plugin\ContextAwarePluginInterface;
+use Drupal\Core\Plugin\PluginFormInterface;
+
+/**
+ * Defines an interface for Offer plugins.
+ */
+interface PromotionOfferInterface extends ExecutableInterface, PluginFormInterface, ConfigurablePluginInterface, PluginInspectionInterface, ContextAwarePluginInterface {
```

Comment:
I'm not sure if it makes sense to extend core's `ExecutableInterface`. It just provides the execute method, but maybe we should provide our own method here that is `apply` or `run`, or something. And it can return a `bool` value so we know if it applied or not.